app/controllers/sessions_controller.rb

```diff
@@ -7,19 +7,51 @@ class SessionsController < Clearance::SessionsController
   def create
     @user = find_user
 
-    if @user&.mfa_enabled?
-      session[:mfa_user] = @user.id
-      render "sessions/otp_prompt"
+    if @user && (@user.mfa_enabled? || @user.webauthn_credentials.any?)
+      setup_webauthn_authentication
+      setup_mfa_authentication
+
+      render "sessions/prompt"
     else
       do_login
     end
   end
 
+  def webauthn_create
+    @user = User.find(session.dig(:webauthn_authentication, "user"))
+    @challenge = session.dig(:webauthn_authentication, "challenge")
```

Comment:
What happens if `webauthn_authentication` is unset or the fields are missing?

Reply:
It responds with a 401 unathorized. This scenario is covered in a test [here](https://github.com/rubygems/rubygems.org/pull/2865/files#diff-3330307a836699182b69bc210bd2b62dbd7ef41c26978ce7ff8e7311bd4629dcR408)

Reply:
Awesome, thank you!